python/bucketbase/ibucket.py

```diff
@@ -328,6 +346,8 @@ def put_object_stream(self, name: PurePosixPath | str, stream: BinaryIO) -> None
         :raises IOError: If stream operations fail
         """
         self._lock_object(name)
+        if self._base_bucket.exists(name):
+            raise io.UnsupportedOperation(f"Object {name} already exists in AppendOnlySynchronizedBucket")
```

Comment:
Here's a BUG, since the Lock will remain opened, as on the previous line the lock is opened, and not closed here.